Mathlib/Topology/JacobsonSpace.lean

```diff
@@ -0,0 +1,172 @@
+      exact ⟨Set.inter_subset_left, Set.inter_subset_right⟩
+    rw [Set.subset_diff, disjoint_self, Set.bot_eq_empty] at this
+    exact hZ this.2
+  · intro H Z hZ
+    refine subset_antisymm (hZ.closure_subset_iff.mpr Set.inter_subset_left) ?_
+    rw [← Set.disjoint_compl_left_iff_subset, Set.disjoint_iff_inter_eq_empty,
+      ← Set.not_nonempty_iff_eq_empty]
+    intro H'
+    have := H _ H' (isClosed_closure.isOpen_compl.isLocallyClosed.inter hZ.isLocallyClosed)
+    rw [Set.nonempty_iff_ne_empty, Set.inter_assoc, ne_eq,
+      ← Set.disjoint_iff_inter_eq_empty, Set.disjoint_compl_left_iff_subset] at this
+    exact this subset_closure
+
+lemma nonempty_inter_closedPoints [JacobsonSpace X] {Z : Set X}
+    (hZ : Z.Nonempty) (hZ' : IsLocallyClosed Z) : (Z ∩ closedPoints X).Nonempty :=
+  jacobsonSpace_iff_locallyClosed.mp inferInstance Z hZ hZ'
+
+lemma isClosed_singleton_of_isLocallyClosed_singleton [JacobsonSpace X] {x : X}
+    (hx : IsLocallyClosed {x}) : IsClosed {x} := by
+  obtain ⟨_, ⟨y, rfl : y = x, rfl⟩, hy'⟩ :=
+    nonempty_inter_closedPoints (Set.singleton_nonempty x) hx
+  exact hy'
+
+lemma IsOpenEmbedding.preimage_closedPoints (hf : IsOpenEmbedding f) [JacobsonSpace Y] :
+    f ⁻¹' closedPoints Y = closedPoints X := by
+  apply subset_antisymm (preimage_closedPoints_subset hf.inj hf.continuous)
+  intros x hx
+  apply isClosed_singleton_of_isLocallyClosed_singleton
+  rw [← Set.image_singleton]
+  exact (hx.isLocallyClosed.image hf.toInducing hf.isOpen_range.isLocallyClosed)
+
+lemma JacobsonSpace.of_isOpenEmbedding [JacobsonSpace Y] (hf : IsOpenEmbedding f) :
+    JacobsonSpace X := by
+  rw [jacobsonSpace_iff_locallyClosed, ← hf.preimage_closedPoints]
+  intros Z hZ hZ'
+  obtain ⟨_, ⟨x, hx, rfl⟩, hx'⟩ := nonempty_inter_closedPoints
+    (hZ.image f) (hZ'.image hf.toInducing hf.isOpen_range.isLocallyClosed)
+  exact ⟨_, hx, hx'⟩
+
+lemma JacobsonSpace.of_isClosedEmbedding [JacobsonSpace Y] (hf : IsClosedEmbedding f) :
+    JacobsonSpace X := by
+  rw [jacobsonSpace_iff_locallyClosed, ← hf.preimage_closedPoints]
+  intros Z hZ hZ'
+  obtain ⟨_, ⟨x, hx, rfl⟩, hx'⟩ := nonempty_inter_closedPoints
+    (hZ.image f) (hZ'.image hf.toInducing hf.isClosed_range.isLocallyClosed)
+  exact ⟨_, hx, hx'⟩
+
+lemma JacobsonSpace.discreteTopology [JacobsonSpace X]
+    (h : (closedPoints X).Finite) : DiscreteTopology X := by
+  have : closedPoints X = Set.univ := by
+    rw [← Set.univ_subset_iff, ← closure_closedPoints,
+      closure_subset_iff_isClosed, ← (closedPoints X).biUnion_of_singleton]
+    exact h.isClosed_biUnion fun _ ↦ id
+  have inst : Finite X := Set.finite_univ_iff.mp (this ▸ h)
+  rw [← forall_open_iff_discrete]
+  intro s
+  rw [← isClosed_compl_iff, ← sᶜ.biUnion_of_singleton]
+  refine sᶜ.toFinite.isClosed_biUnion fun x _ ↦ ?_
+  rw [← mem_closedPoints_iff, this]
+  trivial
+
+instance (priority := 100) [Finite X] [JacobsonSpace X] : DiscreteTopology X :=
+  JacobsonSpace.discreteTopology (Set.toFinite _)
```

Comment:
Could you please also add the instance in the other direction? Discrete => Jacobson

Reply:
Okay I added T2 => Jacobson